### PR TITLE
Add implementation of Exit command

### LINKED ISSUE #238: Add support for Exit command
#### Description:

Add implementation of Exit command added in https://github.com/BHoM/BHoM_Adapter/issues/302

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### NOTE: Depends on 

https://github.com/BHoM/BHoM_Adapter/pull/303

 ### Issues addressed by this PR

 Closes #238 

 

Adding support for the Exit command

 ### Test files

https://burohappold.sharepoint.com/:f:/s/BHoM/EjpTgU6jo_tJr30rbJs0a_gBnZnhI1rYcv3aQ-Yc3A6E2g?e=fenJ0J

 ### Changelog

 ### Additional comments

### CODE REVIEW

**Review by JosefTaylor — DISMISSED** [review]
Looks great, works as expected.

**Comment by JosefTaylor** [comment]
@BHoMBot check required

**Comment by JosefTaylor** [comment]
@BHoMBot check copyright-compliance

**Comment by JosefTaylor** [comment]
@BHoMBot check dataset-compliance

**Comment by JosefTaylor** [comment]
@BHoMBot check ready-to-merge

**Comment by IsakNaslundBh** [comment]
After discussions in https://github.com/BHoM/Robot_Toolkit/pull/466#issuecomment-904922155 Added check for if file path existed before trying to exit if the `SaveBeforeClose` option is true

**Comment by IsakNaslundBh** [comment]
@BHoMBot check required

**Comment by IsakNaslundBh** [comment]
@BHoMBot check required

File: SAP2000_Adapter/AdapterActions/Execute.cs [hunk 1]
```diff
@@ -121,6 +121,25 @@ public bool RunCommand(ClearResults command)
 
         /***************************************************/
 
+        public bool RunCommand(Exit command)
+        {
+            if (command.SaveBeforeClose)
+            {
+                if (m_app.SapModel.GetModelFilepath() == "(Untitled)")
```
Comment (JosefTaylor) [inline SAP2000_Adapter/AdapterActions/Execute.cs:128]:
```suggestion
                if (m_app.SapModel.GetModelFilename() == "(Untitled)")
```

Comment (JosefTaylor) [reply SAP2000_Adapter/AdapterActions/Execute.cs:128]: Need just the file name, not full path. I tested with this change and it works properly, failing to close the file and throwing the desired error in Grasshopper. Without it, the error is never hit, and SAP2000 throws up a dialogue about failing to save, then closes after the user accepts.

**Review by JosefTaylor — CHANGES_REQUESTED** [review]
Almost works!

**Comment by IsakNaslundBh** [comment]
@BHoMBot check required

**Comment by IsakNaslundBh** [comment]
@BHoMBot check compliance

File: SAP2000_Adapter/AdapterActions/Execute.cs (hunk 1 above)
Comment (JosefTaylor) [reply SAP2000_Adapter/AdapterActions/Execute.cs]: GetModelFilepath() returns "" rather than "(Untitled)", cancelling my comment.

**Review by JosefTaylor — APPROVED** [review]

**Comment by JosefTaylor** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**